terraform/modules/apps/frontend/main.tf

```diff
@@ -18,11 +18,109 @@ module "app" {
   service_discovery_namespace_id   = var.service_discovery_namespace_id
   service_discovery_namespace_name = var.service_discovery_namespace_name
   extra_security_groups            = [var.govuk_management_access_security_group]
+  desired_count                    = var.desired_count
+
+  load_balancers = [{
+    target_group_arn = aws_lb_target_group.public.arn
+    container_name   = var.service_name
+    container_port   = 80
+  }]
+}
+
+#
+# Internet-facing load balancer
+#
+
+# TODO: use a single, ACM-managed cert with both domains on. There is already
+# such a cert in integration/staging/prod (but it needs defining in Terraform).
+data "aws_acm_certificate" "public_lb_default" {
+  domain   = "*.test.govuk.digital"
+  statuses = ["ISSUED"]
+}
+
+data "aws_acm_certificate" "public_lb_alternate" {
+  domain   = "*.test.publishing.service.gov.uk"
+  statuses = ["ISSUED"]
+}
+
+resource "aws_lb" "public" {
+  name               = "fargate-public-${var.service_name}"
+  internal           = false
+  load_balancer_type = "application"
+  security_groups    = [aws_security_group.public_alb.id]
+  subnets            = var.public_subnets
+}
+
+resource "aws_lb_target_group" "public" {
+  name        = "${var.service_name}-public"
+  port        = 80
+  protocol    = "HTTP"
+  vpc_id      = var.vpc_id
+  target_type = "ip"
+
+  health_check {
+    path = "/"
+  }
+
+  depends_on = [aws_lb.public]
+}
+
+resource "aws_lb_listener" "public" {
+  load_balancer_arn = aws_lb.public.arn
+  port              = 443
+  protocol          = "HTTPS"
+  ssl_policy        = "ELBSecurityPolicy-2016-08"
+  certificate_arn   = data.aws_acm_certificate.public_lb_default.arn
+
+  default_action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.public.arn
+  }
+}
+
+resource "aws_lb_listener_certificate" "publishing_service" {
+  listener_arn    = aws_lb_listener.public.arn
+  certificate_arn = data.aws_acm_certificate.public_lb_alternate.arn
 }
 
-# TODO: Is this needed?
-resource "aws_security_group" "service" {
-  name        = "fargate_${var.service_name}_ingress"
+resource "aws_security_group" "public_alb" {
+  name        = "fargate_${var.service_name}_public_alb"
   vpc_id      = var.vpc_id
-  description = "Permit internal services to access the ${var.service_name} ECS service"
+  description = "${var.service_name} Internet-facing ALB"
+}
+
+resource "aws_security_group_rule" "public_alb_443_https_office_ips" {
+  type      = "ingress"
+  from_port = 443
+  to_port   = 443
+  protocol  = "tcp"
+
+  security_group_id = aws_security_group.public_alb.id
+  cidr_blocks       = ["213.86.153.212/32", "213.86.153.213/32", "213.86.153.214/32", "213.86.153.235/32", "213.86.153.236/32", "213.86.153.237/32", "85.133.67.244/32"]
```

Comment:
Having to duplicate this list of office networks multiple times seems messy and error prone. I suggest we instead just set a network ACL for the whole test VPC if we want to limit the test environment to the office.

Reply:
Yes, I can do that.

Reply:
Ok, please at least set a local in security_group_rules.tf then because the duplication is unacceptable.

Reply:
Yes, I have done that, I included a variable. Anyway, we will have to refactor in the public alb in the apps because they are duplication themselves, I was copying them straight.